h11/_receivebuffer.py

```diff
@@ -51,56 +28,52 @@ def __bool__(self):
 
     # for @property unprocessed_data
     def __bytes__(self):
-        return bytes(self._data[self._start :])
+        return bytes(self._data)
 
     if sys.version_info[0] < 3:  # version specific: Python 2
         __str__ = __bytes__
         __nonzero__ = __bool__
 
     def __len__(self):
-        return len(self._data) - self._start
-
-    def compress(self):
-        # Heuristic: only compress if it lets us reduce size by a factor
-        # of 2
-        if self._start > len(self._data) // 2:
-            del self._data[: self._start]
-            self._looked_at -= self._start
-            self._start -= self._start
+        return len(self._data)
 
     def __iadd__(self, byteslike):
         self._data += byteslike
         return self
 
     def maybe_extract_at_most(self, count):
-        out = self._data[self._start : self._start + count]
+        out = self._data[:count]
         if not out:
             return None
-        self._start += len(out)
+        # Note that front-truncation of bytesarray is O(1), from Python 3.4
+        # onwards, thanks to some excellent work by Antoine Pitrou:
```

Comment:
I suggest keeping it as "amortized O(1)" , a bit more instructive and technically correct:

```suggestion
        # Note that front-truncation of bytesarray is amortized O(1), from
        # Python 3.4 onwards, thanks to some excellent work by Antoine Pitrou:
```